app/services/distance-list.ts

```diff
@@ -57,6 +60,10 @@ export default class DistanceListService extends Service {
     }
     return this.options;
   }
+
+  getSelectedDistance(id: string): DistanceOption | undefined {
+    return this.options.find((option) => option.id === id);
+  }
```

Comment:
This method should handle the undefined or empty strings that are passed. Saw that you where doing this below
```js
      const distance = this.distanceList.getSelectedDistance(
        this.filterService.asQueryParams.afstand ?? '',
      );
```

```suggestion
  getSelectedDistance(id?: string): DistanceOption | undefined {
    if(!id || id.trim() === '') {
      return undefined; // or null
    }
    return this.options.find((option) => option.id === id);
  }
```

than you don't have to think about what value you are passing as this method handles it